programs/client/Client.cpp

```diff
@@ -1353,6 +1359,9 @@ void Client::readArguments(
             }
             else if (arg.starts_with("--port"))
             {
+                if (has_connection_string)
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Mixing connection string and --host/--port client arguments is prohibited");
```

Comment:
See prev. comment.

Reply:
By the way, we might want to have a connection string and user/pwd/database mutually exclusive. Main reason here is consistency and simplicity with the behavior of --host and --port.

Reply:
Maybe you're right, and we should make the behavior the same for the other options and throw exceptions in case of mixing: 

connection `string+user`
connection `string+password`
connection `string+database`.
And I guess: `connection string+secure`

I don't have a strong opinion on how to do it right. I know that PosgreSQL prohibits that.
If connection string is used in PostgreSQL, no other console arguments are accepted. 

But in our ClickHouse case we can't specify keys like -m, -n, --stage, --progress after client started. We could add those keys to connection string, but they connection string will be used not just for connection.

Reply:
Also we have difference in password behaviour:
`clickhouse-client clickhouse//user:@` // user with empty password = no password argument specified

is not equvalent to
`clickhouse-client --user user --password` // user with `--password` specified without value, the value will be implicitly set to `\n`.

`password "\n"` means ask user for a password. I want add the same behaviour to the connection string:

`clickhouse-client clickhouse//user:@localhost` - ':' is used: ask user for a password
`clickhouse-client clickhouse//user@localhost` - ':' is not used, just try to connect to localhost without password